### PR TITLE
chore: switch from Travis to Github Actions

### PR DESCRIPTION
**Pull request by jdpedrie**
This change removes Travis from the test matrix and replaces it with Github Actions. It also replaces the code style checker with a similar one to what Google Cloud PHP uses. It includes fixes to conform to that style checker.

It generates and pushes documentation to github pages on push to master and branch.

PTAL, LMK, IDK.

Sample run: https://github.com/jdpedrie/google-auth-library-php/actions/runs/67515476
Generated documentation: http://jdpedrie.github.io/google-auth-library-php/master/

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+name: Test Suite
+on: [push, pull_request]
```
Comment (bcoe) [inline .github/workflows/tests.yml:2]: I suggest listening on these events:

```
on:
  push:
    branches:
      - master
  pull_request:
```

This way you don't kick off two sets of tests for a pull_request.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+jobs:
+    test:
+        runs-on: ${{matrix.operating-system}}
+        strategy:
+            matrix:
+                operating-system: [ ubuntu-latest ]
+                php: [ "5.6", "7.0", "7.1", "7.2", "7.3", "7.4" ]
+        name: PHP ${{matrix.php }} Unit Test
+        steps:
+            - uses: actions/checkout@master
```
Comment (bcoe) [inline .github/workflows/tests.yml:13]: I might be tempted to use this at a specific tag, I think we use `v2` for our Node.js repos (just so you don't get any surprise regressions).

File: src/AccessToken.php [hunk 3]
```diff
@@ -82,22 +82,19 @@ public function __construct(
...
+     *        "federated_signon_certs_v3".
+     * @param string $options.certsLocation The location (remote or local) from which
      *        to retrieve certificates, if not cached. This value should only be
      *        provided in limited circumstances in which you are sure of the
      *        behavior.
-     *     @type string $cacheKey The cache key of the cached certs. Defaults to
-     *        the sha1 of $certsLocation if provided, otherwise is set to
-     *        "federated_signon_certs_v3".
-     *     @type bool $throwException Whether the function should throw an
+     * @param bool $options.throwException Whether the function should throw an
```
Comment (bshaffer) [inline src/AccessToken.php:95]: This is different than the style we use for `google/cloud`. Are we moving to this format over there as well?

File: src/OAuth2.php [hunk 4]
```diff
@@ -766,7 +770,7 @@ public function setRedirectUri($uri)
      */
     public function getScope()
     {
-        if (is_null($this->scope)) {
+        if (is_null($this->scope) || is_string($this->scope)) {
```
Comment (bshaffer) [inline src/OAuth2.php:773]: The typehint is wrong, the `$scope` property should always be an array. What prompted this change?

File: src/AccessToken.php (hunk 3 above)
Comment (jdpedrie) [reply src/AccessToken.php:95]: The documentation generator used doesn't support the style we use on google cloud php. I wouldn't be opposed to moving to this style there as well, but it's a bigger job and would require changes to the generated code as well.

File: src/OAuth2.php (hunk 4 above)
Comment (jdpedrie) [reply src/OAuth2.php]: fixed (and fixed the property type)

**Pull request merged**